internal/pkg/middleware/authororization.go

```diff
@@ -0,0 +1,74 @@
+package middleware
+
+import (
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/TheFlies/ofriends/internal/pkg/glog"
+	"github.com/TheFlies/ofriends/internal/pkg/jwt"
+	"github.com/TheFlies/ofriends/internal/pkg/respond"
+)
+
+const (
+	none      = 1
+	roleUser  = 2
+	roleAdmin = 3
+)
+
+func Authorization(roles ...int) func(http.HandlerFunc) http.HandlerFunc {
+	l := glog.New().WithField("package", "middleware")
+	if len(roles) == 0 {
+		l.Errorf("role list must not blank")
+		return nil
+	}
+	sort.SliceIsSorted(roles, func(i, j int) bool {
+		return i > j
+	})
+	l.Infof("list role %v", roles)
+	return func(next http.HandlerFunc) http.HandlerFunc {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if ok := RoleCheck(r, roles...); !ok {
+				l.Infof("authorization fail")
+				respond.JSON(w, http.StatusUnauthorized, map[string]string{"status": "401", "message": "authorization"})
+				return
+			}
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+func RoleCheck(r *http.Request, roles ...int) bool {
```

Comment:
We don't have `Moderator` role for now, but let assume that `NormalUser < Moderator < Admin`.
What if I want to allow `Admin` and `NormalUser` to access this API, but not `Moderator`, so in that case, does this func still run correctly?

Reply:
No ! When I put a additional role, which is `Moderator` this func is not correct, so I think authorization must define in detail who can do what
Example :
`middleware.Authorization(roleUser)` just user have permission, `admin` and `moderator` haven't permission
`middleware.Authorization(roleAdmin)` just admin have permission 
if you